src/txdb.cpp

```diff
@@ -452,6 +673,16 @@ bool CBlockTreeDB::ReadFlag(const std::string &name, bool &fValue) {
     return true;
 }
 
+bool CBlockTreeDB::blockOnchainActive(const uint256 &hash) {
+    CBlockIndex* pblockindex = mapBlockIndex[hash];
```

Comment:
We should use `BlockMap::iterator mi = mapBlockIndex.find(hash)` and `if (mi != mapBlockIndex.end())` , otherwise in case of unknown hash we would add new (null) entries in the map.

Reply:
Fixed.